TestResultSummaryService/parsers/BenchmarkParser.js

```diff
@@ -2,92 +2,53 @@ const Parser = require( './Parser' );
 const BenchmarkMetricRouter = require( './BenchmarkMetricRouter' );
 const BenchmarkMetric = require( './BenchmarkMetric' );
 
-const benchmarkDelimiterRegex = /\*\*\*\*\*\*\*\*\*\* START OF NEW TESTCI BENCHMARK JOB \*\*\*\*\*\*\*\*\*\*[\r\n]/;
-const benchmarkNameRegex = /Benchmark Name: (.*) Benchmark Variant: .*[\r\n]/;
-const benchmarkVariantRegex = /Benchmark Name: .* Benchmark Variant: (.*)[\r\n]/;
-const productResourceRegex = /Product Resource: (.*)[\r\n]/;
-
 class BenchmarkParser extends Parser {
 
-    // True iff all benchmarks and their corresponding iterations in the given run can be parsed
-    static canParse( buildName, output ) {
-        return benchmarkDelimiterRegex.test(output);
+    static getBenchmarkInfo( testName ) {
+        let result, benchmarkName, benchmarkVariant;
+        if (testName) {
+            if ((result = testName.split("_")) != null) {
+                benchmarkName = result[0];
+                benchmarkVariant = result[1];
+            }
+        }
+        return {benchmarkName, benchmarkVariant};
     }
 
-    // Returns an iterator object which accesses each benchmark iteration from the given output
-    static getBenchmarkIterator(output) {
-        let index = 1;
-        let splitOutput = output.split(benchmarkDelimiterRegex);
-        let splitOutputLength = splitOutput.length;
-    
-        // No delimiter inside the output
-        if (!(Array.isArray(splitOutput)) || splitOutputLength === 1) {
-            return null;
+    // True iff all benchmarks and their corresponding iterations in the given run can be parsed
+    static canParse( testName ) {
+        const {benchmarkName, benchmarkVariant} = this.getBenchmarkInfo(testName);
+        if(benchmarkName && benchmarkVariant && BenchmarkMetricRouter[benchmarkName] && BenchmarkMetricRouter[benchmarkName][benchmarkVariant]) {
+            return true;
+        } else if ( isNaN (benchmarkVariant) && benchmarkName && benchmarkVariant && BenchmarkMetricRouter[benchmarkName]["0"]) {
```

Comment:
`isNaN (benchmarkVariant)` won't work. This would even return true for something that's not there (i.e. variant = "t") and it would go to "0". We should go to "0" only if the all characters inside the variant string are numbers.

Reply:
We also need the check `BenchmarkMetricRouter[benchmarkName]` again since we will incorrectly return true for even a test job.